ruby/src/main/java/com/google/protobuf/jruby/RubyRepeatedField.java

```diff
@@ -117,12 +117,19 @@ public IRubyObject indexSet(ThreadContext context, IRubyObject index, IRubyObjec
      * call-seq:
      *     RepeatedField.[](index) => value
      *
-     * Accesses the element at the given index. Throws an exception on out-of-bounds
-     * errors.
+     * Accesses the element at the given index. Returns nil on out-of-bounds
      */
     @JRubyMethod(name = "[]")
     public IRubyObject index(ThreadContext context, IRubyObject index) {
-        return this.storage.eltInternal(RubyNumeric.num2int(index));
+        int arrIndex = RubyNumeric.num2int(index);
+        int arrSize = this.storage.size();
+        if (arrIndex < 0 && arrSize > 0) {
+            arrIndex = arrSize + arrIndex;
+        }
+        if (arrIndex < 0 || arrIndex >= arrSize) {
+            return context.runtime.getNil();
+        }
+        return this.storage.eltInternal(arrIndex);
```

Comment:
this is functional but smells... if you agree, I can pull it out into a helper, like in the C file